gigascience/gigadb-website
# Add disk usage calculator and Gitter notifier using swatch

# Pull request for issue: 

This is a pull request for the following functionalities:

* Setup a disk usage monitoring and alerting for the bastion server

## How to test?

After building an AWS environment for staging, 
Log in to the bastion with SSH, and run the following command:

```
$ swatch -c check-swatch.conf -p ./diskUsage
```

Then you should see a message in the Gitter notification channel showing disk usage alongside a **(drill)** flag.

## How have functionalities been implemented?

There are three moving pieces to the workflow:

1. finding out disk usage
2. notifying that value to the Gitter notification channel if the value is above a threshold
3. setup a cronjob to have that health check happen daily on bastion (no need to do more frequently, as disk usage increase is mostly the result of daily cronjob).


### Details

1. ``df`` command will show information about disk usage, we pipe it through various command to extract just the value for the EBS volume (the one whose path contain ``/dev/``)
2. Install swatch and:
	a. wraps a curl call to the Gitter api into a notify script. It needs Gitter room id, gitter api token and the GigaDb environment variable 
	b. configure ``swatch.conf`` with different thresholds (< 50 %: green, no notifcation; < 70 %: yellow, notify ; > 70 %: red, notify)
	c. call swatch using the ``-p`` argument as we want to pipe the value of disk usage to the notify script
3. Setting up the cronjob, require updating bastion playbook to do: 

	a. Create the diskUsage monitoring script (it does 1.)
	b. Create the Gitter notify script (it does 2.a.)
	c. Copy swatch config file (that's 2.b. as well as a test config for drills)
	d. Install swatch package (use ``dnf`` to install the appropriate RPM)
	e. Setup cronjob to automate daily disk usage monitoring


Finally, other files were changed to enable the environment variables required for 2.a. to be available in the ``.env`` file on the bastion server.


## Any changes to documentation?

Added two new variables to the **Forks** section of: 

* docs/variables.md
